src/server/server.ts

```diff
@@ -156,36 +156,24 @@ namespace ts.server {
     var logger = createLoggerFromEnv();
 
     let pending: string[] = [];
-    function queueMessage(s: string) {
-        pending.push(s);
-        if (pending.length === 1) {
-            drain();
-        }
-    }
-
-    function drain() {
-        Debug.assert(pending.length > 0);
-        writeBuffer(new Buffer(pending[0], "utf8"), 0);
-    }
-
-    function writeBuffer(buffer: Buffer, offset: number) {
-        const toWrite = buffer.length - offset;
-        fs.write(1, buffer, offset, toWrite, undefined, (err, written, buffer) => {
-            if (toWrite > written) {
-                writeBuffer(buffer, offset + written);
-            }
-            else {
-                Debug.assert(pending.length > 0);
-                pending.shift();
-                if (pending.length > 0) {
-                    drain();
+    let canWrite = true;
+    function writeMessage(s: string) {
+        if (!canWrite) {
+            pending.push(s);
+        }
+        else {
+            canWrite = false;
+            process.stdout.write(new Buffer(s, "utf8"), () => {

```

Comment:
this is the common practice and standard buffer API. In theory we can pool then though in practice I'll start doing this only if I have convincing numbers